client/onboarding/steps/embedded-kyc.tsx

```diff
@@ -41,104 +30,30 @@ const EmbeddedKyc: React.FC< Props > = ( {
-		( () => Promise< string > ) | null
-	>( null );
-	const [
-		stripeConnectInstance,
-		setStripeConnectInstance,
-	] = useState< StripeConnectInstance | null >( null );
 	const [ loading, setLoading ] = useState( true );
 	const [ finalizingAccount, setFinalizingAccount ] = useState( false );
 	const [ loadErrorMessage, setLoadErrorMessage ] = useState( '' );
 
-	const fetchAccountSession = useCallback( async () => {
-		try {
-			const isEligible = ! continueKyc && ( await isPoEligible( data ) );
-			const accountSession = await createAccountSession(
-				data,
-				isEligible
-			);
-			if ( accountSession && accountSession.clientSecret ) {
-				trackRedirected( isEligible, true );
-				return accountSession; // Return the full account session object
-			}
-
-			setLoadErrorMessage(
-				__(
-					"Failed to create account session. Please check that you're using the latest version of WooPayments.",
-					'woocommerce-payments'
-				)
-			);
-		} catch ( error ) {
-			setLoadErrorMessage(
-				__(
-					'Failed to retrieve account session. Please try again later.',
-					'woocommerce-payments'
-				)
-			);
-		}
-
-		// Return null if an error occurred.
-		return null;
-	}, [ continueKyc, data ] );
-
-	// Function to fetch clientSecret for use in Stripe auto-refresh or initialization
-	const fetchClientSecret = useCallback( async () => {
-		const accountSession = await fetchAccountSession();
-		if ( accountSession ) {
-			return accountSession.clientSecret; // Only return the clientSecret
-		}
-		throw new Error( 'Error fetching the client secret' );
-	}, [ fetchAccountSession ] );
-
-	// Effect to fetch the publishable key and clientSecret on initial render
-	useEffect( () => {
-		const fetchKeys = async () => {
-			try {
-				const accountSession = await fetchAccountSession();
-				if ( accountSession ) {
-					setLocale( accountSession.locale );
-					setPublishableKey( accountSession.publishableKey );
-					setClientSecret( () => fetchClientSecret );
-				}
-			} catch ( error ) {
-				setLoadErrorMessage(
-					__(
-						'Failed to create account session. Please check that you are using the latest version of WooPayments.',
-						'woocommerce-payments'
-					)
-				);
-			}
-		};
-
-		fetchKeys();
-	}, [ data, continueKyc, fetchAccountSession, fetchClientSecret ] );
-
-	// Effect to initialize the Stripe Connect instance once publishableKey and clientSecret are ready.
-	useEffect( () => {
-		if ( publishableKey && clientSecret && ! stripeConnectInstance ) {
-			const stripeInstance = loadConnectAndInitialize( {
-				publishableKey,
-				fetchClientSecret,
-				appearance: {
-					overlays: 'drawer',
-					variables: appearance.variables,
-				},
-				locale: locale.replace( '_', '-' ),
-			} );
-
-			setStripeConnectInstance( stripeInstance );
-		}
-	}, [
-		publishableKey,
-		clientSecret,
-		stripeConnectInstance,
-		fetchClientSecret,
-		locale,
-	] );
+	/**
+	 * This is a custom hook that retrieve the account session data.
```

Comment:
This documentation is great, but IMO it make more sense for it to be in the `client/utils/embedded-components/account-session.ts` file, since we might use it in multiple places in future?

Reply:
Moving the docs to where the hook is defined definitely makes more sense. Thanks!